### PR TITLE
Add formatting for i18n attributes.

### LINKED ISSUE #5875: [Angular] Format values of i18n attributes
**Prettier 1.16.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAFgRgAQEtMAcUAvADogAq6uAznnQIbYBucATgJ7YA20A5nljsoDGLmgNuuAF6jxUbABM4NMG1wAHMdGzwAHjAA02OHrBw4i3FAEx0cbJCgAzXHwCubS9g3rY2AHdcRTtjACN3GEcId25FEwYaXG4uMIdnCDYAW1F4RXIAPlIFbAAJOG5eAEJi1AB6LCKoVCyOAFowdxoYCCyO3o1oBCjO7t6AQRgYNjIQLIsNcjwCTCgOrp6syenZqlp6bCZWTh5+Q6n1CPg2sMTva3g2EW0RKVkXpRU1TQ-9I0DqGB0IduDQICYzBZFHQ7A4nK4PF54r4HoFgnZCrU6q11mM+pAsoMoMMCiBDCAIFp5DRkKAGGw2BAAgAFekIGkoBjMCDBMkgMJsBhgADWcBgAGUNELrHxkNN3HByegYFluAB1ajwGhS8zi9m4MTMA0cZDgGg08nWGjsGDMwV8HLIZySa3kgBWND0ACFBSKxeKGPMADLWOBOl2KkAevTimXcOAARXcEHg4dBkalbGtbFNDBssXpfJRsDV6PQyAIAAZyb4INa1YKNKbfCp2Kw+V4AI7uXBeO0MB0MNOukDWrK4OVsBXkpI2eNJlNhpDO9PkmAMMKlkLlpAAJjXguSMoAwr1HSgoEM+V04BQNxyV9aAL5PoA)
```sh
--parser angular
```

**Input:**
```html
<h1 i18n="This is a very long internationalization description text, exceeding the configured print width, but could easily be formatted">
  Hello!
</h1>
<my-custom-component customAttr="meep" i81n-customAttr="This is a very long attribute-based internationalization description text, which also exceeds the configured print width">
</my-custom-component>
```

**Output:**
```html
<h1
  i18n="This is a very long internationalization description text, exceeding the configured print width, but could easily be formatted"
>
  Hello!
</h1>
<my-custom-component
  customAttr="meep"
  i81n-customAttr="This is a very long attribute-based internationalization description text, which also exceeds the configured print width"
>
</my-custom-component>

```

**Expected behavior:**
Currently, Prettier doesn't wrap the `i18n` attribute content. Based on discussions in #5479 and #5482, I think this would be a good candidate for formatting, since the `i18n` content is well-known, and tends to be long (and thus is more of a pain when unformatted).

Documentation on Angular's `i18n` attributes, for reference: https://angular.io/guide/i18n

Ideally, I'd want output similar to *either* this (with a continuation indent):

```html
<h1
  i18n="This is a very long internationalization description text, exceeding the
    configured print width, but could easily be formatted"
>
  Hello!
</h1>
<my-custom-component
  customAttr="meep"
  i81n-customAttr="This is a very long attribute-based internationalization
    description text, which also exceeds the configured print width"
>
</my-custom-component>

```

*Or* this (with an alignment to the opening quote, `"`):

```html
<h1
  i18n="This is a very long internationalization description text, exceeding the
        configured print width, but could easily be formatted"
>
  Hello!
</h1>
<my-custom-component
  customAttr="meep"
  i81n-customAttr="This is a very long attribute-based internationalization
                   description text, which also exceeds the configured print
                   width"
>
</my-custom-component>

```

I'd somewhat prefer the continuation indent instead of the quote alignment, as it minimizes wasted horizontal space and seems to be more consistent with how Prettier formats long sequences of attributes.

Does this sound reasonable? If so, I may try sending a PR.

### PR DESCRIPTION
**Pull request by voithos**
Previously, Prettier would largely ignore i18n attributes, not even
wrapping their content, which wasn't ideal since i18n descriptive text
can sometimes get long.

After this, Prettier will auto-wrap the contents of i18n attributes once
they exceed the line length.

Fixes #5875.

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: CHANGELOG.unreleased.md [hunk 1]
```diff
@@ -1014,6 +1014,32 @@ class A {
...
+  i18n="This is a very long internationalization description text, exceeding the configured print width"
+>
+  Hello!
+</h1>
+
+// Output (Prettier master)
+<h1
+  i18n="
+    This is a very long internationalization description text, exceeding the
+    configured print width"
```
Comment (thorn0) [inline CHANGELOG.unreleased.md]: This sample snippet isn't accurate. Please fix it.
1) it's HTML, not TS (` ```ts ` → ` ```html `, format of comments)
2) Your implementation puts the closing quote on a new line, which is not reflected here.

Comment (voithos) [reply CHANGELOG.unreleased.md]: Ah, thank you! Sorry, not sure why I missed these.

Done, on both accounts.

File: src/language-html/printer-html.js [hunk 2]
```diff
@@ -1033,6 +1033,11 @@ function printEmbeddedAttributeValue(node, originalTextToDoc, options) {
      *     bindon-target="angularExpression"
      */
     const ngExpressionBindingPatterns = ["^\\[.+\\]$", "^bind(on)?-"];
+    /**
+     *     i18n="longDescription"
+     *     i18n-attr="longDescription"
```
Comment (lydell) [inline src/language-html/printer-html.js]: I’m not familiar with Angular. Do these kinds of attributes only contain plain text, or could they also contain expressions or code of some sort? If so, we should add a test for that.

Comment (voithos) [reply src/language-html/printer-html.js]: That's a great point - they don't contain code, but they can contain ID syntax as well as "meaning" syntax, denoted by `@@` and `|` separators. Added some test cases.

**Comment by thorn0** [comment]
What about [ICU expressions](https://angular.io/guide/i18n#pluralization)? They're used not inside the attribute, but it would be great if Prettier didn't crash on them if we're going to claim that we support Angular's i18n syntax. If you don't feel like fixing them in this PR, it's okay. I'm just mentioning it here for the sake of completeness.

**Prettier pr-6695**
[Playground link](https://deploy-preview-6695--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAeAzgBwIZQAQCWAjABxQB8AqpgCbbw17AC2BUArvOgDR6YA27AE7Z+vALwAGJgCt26GHigQA7gF884ok2hw8rDvDzYA5hA0QYACzhCmwFm05x0ajQefpjZt6gD0WLjkINwgEJgwBNDoyKDYQkKqAArxCDEo2ABuEAQ0ISAARiJgANZwMADKOGBsJsgwQuxwoVYwzPwA6lYEXNVwFWk9BJk9AJ7I4OgxoWzotjBJIibM2MgAZqJzoTLoAB4AQsVlldjMcAAybHDrm80gO7sVtfxwAIrsltdIG-xbIDhCOZCCa4EzsfjxfKYIRsGAdXLWZAkSShaEQOYdESYCbQly2TLXUIARw+8EW4XSIGw6AAtFA4HAaIz8kI4CSCKzFqYVjdfnc5qx6o1+c83qSvj8-jBsAV4TREUgAEyhBrYAj8WoAYQgzB5KGU9Py8jgABUZelJXdMk0AJJQJmwCpgGERACC9oqMFGL15czcQA)
```sh
--parser angular
```

**Input:**
```html
<span i18n>Updated {minutes, plural, =0 {just now} =1 {one minute ago} other {{{minutes}} minutes ago}}</span>
```

**Output:**
```html
SyntaxError: Missing expected : at the end of the expression [{{{minutes}}] in angular-estree-parser
    at o (https://deploy-preview-6695--prettier.netlify.com/lib/parser-angular.js:15:16462)
    at t.parseNgInterpolation (https://deploy-preview-6695--prettier.netlify.com/lib/parser-angular.js:15:17592)
    at n (https://deploy-preview-6695--prettier.netlify.com/lib/parser-angular.js:15:28473)
    at Object.t.parseInterpolation (https://deploy-preview-6695--prettier.netlify.com/lib/parser-angular.js:15:28811)
    at https://deploy-preview-6695--prettier.netlify.com/lib/parser-angular.js:15:29554
    at Object.parse (https://deploy-preview-6695--prettier.netlify.com/lib/parser-angular.js:15:29211)
    at Object.parse (https://deploy-preview-6695--prettier.netlify.com/lib/standalone.js:12913:21)
    at textToDoc (https://deploy-preview-6695--prettier.netlify.com/lib/standalone.js:15787:25)
    at https://deploy-preview-6695--prettier.netlify.com/lib/standalone.js:15774:16
    at Object.embed$2 [as embed] (https://deploy-preview-6695--prettier.netlify.com/lib/standalone.js:21594:57)
```

**Comment by voithos** [comment]
@thorn0 Thanks for mentioning ICU expressions! I notice something odd, though - I actually just tried testing that code snippet (among other ones) on the master branch both via `./bin/prettier.js` in dev mode, as well as by building prettier and running the dist code. In both cases, they were able to parse and format the ICU expressions just fine. I'm not sure why the playground would be any different - or perhaps I'm just not testing it right locally?

If you have a local repro for that parsing error, could you file a new issue with instructions?

**Comment by thorn0** [comment]
Ah, okay. It's a known issue with the playground. Parser errors in embedded syntaxes (e.g. Angular expression language inside HTML) are normally suppressed and the embedded code simply stays as is. But the playground shows those errors (for debugging purposes, I guess), and it's easy to confuse them with a full crash of Prettier, which I did.

**Pull request merged**